content/docs/user-guide/programming/components/reflection/behavior-context.md

```diff
@@ -0,0 +1,461 @@
++ **Userdata** - Allows you to provide a pointer to user data. The pointer is accessible from all callbacks (like a custom allocator) that you implement for the class.
++ **Method** - Reflects a C++ class function. The first argument is the class pointer. This is the same usage as global methods.
++ **Property** - Reflects class data. The first argument is the class pointer. This is the same usage as global properties.
++ **Enum** - Enums are read-only `int` properties.
++ **Constant** - Constants are read-only properties.
+
+### Attributes
+
+You can use the following, optional built-in attributes to decorate a class.
+
+| Attribute | Description | Type | Values |
+| --- | --- | --- | --- |
+| **Category** | Used by the editor to categorize the object in a list. You can use the forward slash (`/`) separator to nest categories: <br> `Attribute(AZ::Script::Attributes::Category, "Gameplay/Triggers")` | `string` | |
+| **ExcludeFrom** | An optional flag that hides this object from editor lists, self-documentation, preview builds, or all of the above. This flag is primarily used for internal objects that are not intended to be accessible by script. | `AZ::Script::Attributes::ExcludeFlags` | `List`, <br> `Documentation`, <br> `Preview`, <br> `All` |
+| **Storage** | Specifies the owner of the memory storage for the reflected object. Possible values include the script system, runtime native code, and stored by value in the script system's virtual machine (VM). | `AZ::Script::Attributes::StorageType` | `ScriptOwn`, <br> `RuntimeOwn`, <br> `Value` |
+| **ConstructibleFromNil** | Specifies whether the class is constructed by default when nil is provided. | `bool` | `true`, <br> `false` |
+| **ClassNameOverride** | Provides a custom name for script reflection that is different from the behavior context name. | `string` | |
+| **Ignore** | Specifies whether the element is ignored during reflection. | `bool` | `true`, <br> `false` |
+| **Deprecated** | Marks a reflected class, method, EBus, or property as deprecated. | `bool` | `true`, <br> `false` |
+| **ToolTip** | Used by the editor to display additional information in a tooltip. | `string` | |
+
+### Lua usage examples
+
+Classes reflected for scripting become objects that can be instanced in Lua:
+
+```lua
+local myObj = MyClass()
+```
+
+To use non-default constructors in Lua, first reflect the constructor to the behavior context:
+
+```cpp
+behaviorContext->Class<MyClass>("MyClass")
+    ->Constructor<int>();
+```
+
+Then you can instantiate in Lua:
+
+```lua
+local myClass = MyClass(10)
+```
+
+### Using nested classes
+
+To bind a nested class to the behavior context, you must do it from within a function on the nested class. C++ rules about nested class member access from outside scopes make this requirement necessary.
+
+The following counterexample shows a _poorly_ formed nested class. The code does not work.
```

Comment:
If you're going to lead with the bad example, describe a bit more in the code itself via comments (or write some frontmatter/backmatter for the example) showing why it doesn't work, so that the correct code path can be emphasized later.

In general I'm wary about putting "don't do this" bad examples in the docs, but if this would be a common user mistake, no reason not to.

Reply:
So...after taking a closer look at this "bad example" imported from the LY documentation, I couldn't say for sure what was actually "bad" about it. I decided to test it in a project and lo and behold, the "bad example" worked just fine! I ran it by an LY engineer and he, too, agreed there was nothing wrong with it. I don't know what the original contributor of this code was thinking, but I believe that I have proven it to be incorrect. I'm going to remove this section on nested classes.